tests/product/presto_installer.py

```diff
@@ -0,0 +1,168 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Module for installing presto on a cluster.
+"""
+
+import fnmatch
+import os
+import urllib
+
+import prestoadmin
+from tests.product.constants import LOCAL_RESOURCES_DIR
+
+from tests.docker_cluster import DockerCluster
+from tests.product.prestoadmin_installer import PrestoadminInstaller
+from tests.product.topology_installer import TopologyInstaller
+
+RPM_BASENAME = r'presto.*'
+PRESTO_RPM_GLOB = r'presto*.rpm'
+
+DUMMY_RPM_NAME = 'dummy-rpm.rpm'
+
+
+class PrestoInstaller(object):
+
+    def __init__(self, testcase):
+        self.presto_rpm_filename = self._detect_presto_rpm()
+        self.testcase = testcase
+
+    @staticmethod
+    def get_dependencies():
+        return [PrestoadminInstaller, TopologyInstaller]
+
+    def install(self, dummy=False, extra_configs=None):
+        cluster = self.testcase.cluster
+        if dummy:
+            rpm_dir = LOCAL_RESOURCES_DIR
+            rpm_name = DUMMY_RPM_NAME
+        else:
+            rpm_dir = prestoadmin.main_dir
+            rpm_name = self.presto_rpm_filename
+
+        rpm_name = self.copy_presto_rpm_to_master(rpm_dir=rpm_dir,
+                                                  rpm_name=rpm_name,
+                                                  cluster=cluster)
+
+        self.testcase.write_test_configs(cluster, extra_configs)
+        cmd_output = self.testcase.run_prestoadmin(
+            'server install ' +
+            os.path.join(cluster.mount_dir, rpm_name),
+            cluster=cluster
+        )
+
+        self.testcase.default_keywords.update(self.get_keywords(rpm_name))
+
+        return cmd_output
+
+    def get_keywords(self, rpm_name):
+        return {
+            'rpm': rpm_name,
+            'rpm_basename': RPM_BASENAME,
+            'rpm_basename_without_arch': self.presto_rpm_filename[:-11],
+        }
+
+    @staticmethod
+    def assert_installed(testcase, container, msg=None, cluster=None):
+        # cluster keyword arg supports configurable cluster, which needs to
+        # assert that presto isn't installed before testcase.cluster is set.
+        if not cluster:
+            cluster = testcase.cluster
+
+        try:
+            check_rpm = cluster.exec_cmd_on_host(
+                container, 'rpm -q presto-server-rpm')
+            testcase.assertRegexpMatches(
+                check_rpm, RPM_BASENAME + '\n', msg=msg
+            )
+        except OSError as e:
+            if isinstance(testcase.cluster, DockerCluster):
```

Comment:
we can probably remove this commit, @mattsfuller probably found the underlying cause of presto-admin failures